GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java

```diff
@@ -65,9 +68,37 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
             generateNotifications();
         } else if ("purgeDeviceFileJobQueueRecords".equals(action)) {
             purgeDeviceFileJobQueueRecords();
+        } else if ("purgeReportRecords".equals(action)) {
+            purgeReportRecords();
         }
     }
 
+    /**
+     * scans for and deletes Report entries that are either more than one year old
+     */
+    private void purgeReportRecords() {
+        Calendar deadline = Calendar.getInstance();
+        deadline.add(Calendar.YEAR, ONE_YEAR_AGO);
+        log.info("Starting scan for Report entries, older than: " + deadline.getTime());
+        ReportDao reportDao = new ReportDao();
+        List<Report> reportList = reportDao.list("all"); //TODO: only list old entities
```

Comment:
Rather than retrieving all reports, can we have a utility function to list and filter by dates instead?

Reply:
Yes. Need a db index for that, then.